libs/blocks/global-navigation/utilities/utilities.js

```diff
@@ -247,7 +247,11 @@ export const [hasActiveLink, setActiveLink, getActiveLink] = (() => {
 
       [`${origin}${pathname}`, pathname].forEach((path) => {
         if (activeLink) return;
-        activeLink = area.querySelector(`a[href = '${path}'], a[href ^= '${path}?'], a[href ^= '${path}#']`);
+        const linkElements = area.querySelectorAll('a:not([data-modal-hash])');
+        activeLink = [...linkElements].find((el) => {
+          const regex = new RegExp(`^${path}(\\?[^#]*)?(#.*)?$`);
+          return regex.test(el.href);
+        });
```

Comment:
This could be simplified if `linkElements` and `regex` are not cached:

```
activeLink = [...area.querySelectorAll('a:not([data-modal-hash])')].find((el) => {
    return new RegExp(`^${path}(\\?[^#]*)?(#.*)?$`).test(el.href);
});
```

Reply:
And even a little further, since it doesn't need the return for the inner arrow function
```
activeLink = [...area.querySelectorAll('a:not([data-modal-hash])')].find((el) => new RegExp(`^${path}(\\?[^#]*)?(#.*)?$`).test(el.href));
```